EIPS/eip-00.md

```diff
@@ -0,0 +1,789 @@
+
+```
+
+keccak256(AA_TX_TYPE || 0x00 || rlp(transaction_payload)
+
+```
+
+Note that the `chainId` and `accessList` parameters are included in the transaction hash calculation but are not
+available on-chain as part of the `TransactionType4` struct.
+
+### Accepting EOA account as `sender` to achieve native gas abstraction
+
+In case the `sender` address does not have any code deployed and the `deployerData` length is zero,
+interpret the `signature` parameter as `(y_parity, r, s)` and the `nonce` parameter
+as `(validUntil, validAfter, nonce)`.
+Replace the sender validation frame with default ECDSA signature validation.
+Also check the block timestamp is within the `[validUntil, validAfter]` range.
+
+The base transaction gas cost, in this case, is increased by `AA_ECRECOVER_COST`.
+
+The `callData` parameter in this case is interpreted as following:
+
+```
+
+target || value || data
+
+```
+
+### Execution layer block validation
+
+When validating a block, the validity conditions for a block are extended as follows:
+
+```
+
+for txIndex := 0; txIndex < range block.Transactions.Len(); txIndex++ {
+
+    // 1. Save the current transaction
+    txCurr = block.Transactions[txIndex]
+
+    if (txCurr.Type() == AccountAbstractionTransaction) {
+
+      // 2. Start running validations for AA transactions
+      for j := txIndex; j < range block.Transactions().Len(); j++ {
+        tx = block.Transactions[j]
+
+        // 3. Stop after encountering a non-AA transaction (or reaching the end of the block)
+        if (tx.Type() != AccountAbstractionTransaction) {
+          break
+        }
+        context[j], paymasterValidationGasUsed[j], error := validateAccountAbstractionTransaction(tx)
+        assert error == nil
+      }
+
+      // 4. If all validations are successful, go back to the saved tx index and run all executions
+      for j := txIndex; j < range block.Transactions().Len(); j++ {
+        tx = block.Transactions[j]
+        if (tx.Type() != AccountAbstractionTransaction) {
+          break
+        }
+
+        calldataCost := calculateCalldataCost(tx)
+        retCall, error := evm.Call(
+            from: AA_ENTRY_POINT,
+            to: sender,
+            input: callData,
+            gas: callGasLimit - calldataCost)
+
+        txIndex := j // transaction executed - no need to revisit in the outer loop
+
+
+        // 5. Run cleanups if necessary
+        if (context[j].Len() == 0){
+          continue
+        }
+
+        paymasterPostTransactionInput := ABI.encodeWithSelector('postPaymasterTransaction', success, actualGasCost, context[j])
+        retPostTransaction, error := evm.Call(
+            from: AA_ENTRY_POINT,
+            to: paymaster,
+            input: paymasterPostTransactionInput,
+            gas: paymasterGasLimit - paymasterValidationGasUsed[j])
+      }
+   }
+   else {
+      // handle other types of transactions
+      evm.Apply(txCurr)
+   }
+}
+
+```
+
+### RPC methods (eth namespace)
```

Comment:
these changes should be made to `ethereum/execution-apis`

Reply:
Right, they belong there, but since this EIP currently targets L2 networks, would it be appropriate to add RPC methods to the ethereum doc despite not being supported by L1?  Or should we clone ethereum/execution-apis to a new execution-apis-l2 repo?

Reply:
I would _love_ if each L2 maintained their own fork of `ethereum/execution-apis` that highlights the differences vs the L1 API.

I think I'd be useful to see these RPC methods on a fork.

Reply:
But until they do, where should this API live if not in the EIP?  Should we start a fork of `ethereum/execution-apis` ourselves for this?